controllers/sync/eventMapper.go

```diff
@@ -13,7 +13,11 @@ import (
 )
 
 func eventMapper(obj client.Object) []reconcile.Request {
-	event := obj.(*corev1.Event)
+	event, eventOK := obj.(*corev1.Event)
+	if !eventOK {
+		panic(eventOK)
+	}
```

Comment:
I think `if !ok {panic}` is basically the current behavior if the assertion isn't checked and fails. If we're confident the assertion won't fail, I'd rather have a comment to ignore the lint on this line.

```
//nolint:forcetypeassert
```

Reply:
Or, looking at other cases like this in the PR, should it return an empty slice back?

Reply:
@JustinKuli I think the second suggestion may be a better idea, I will polish the log here in another logger pr

Reply:
@JustinKuli It seems an empty slice will false fail test cases, not look good to me on the corner case handling. I ignored this lint here and will take a further looking if need to open a new pr for it.

Reply:
Awesome, thanks for checking!